inference-engine/src/vpu/graph_transformer/src/middleend/passes/adjust_data_batch.cpp

```diff
@@ -94,7 +94,9 @@ void PassImpl::run(const Model& model) {
         auto curSubGraph = extractNextSubGraph(stagesToSplit);
         IE_ASSERT(!curSubGraph.empty());
 
-        if (curSubGraph.size() <= s_subgraphSizeThreshold) {
+        bool isStaticBatch = (curSubGraph.front()->attrs().get<Data>("batchData")->parentDataToShapeEdge() == nullptr);
```

Comment:
Unfortunately, practice to keep `vpu::Handle` to `vpu::Data` or `vpu::Stage` is a bad practice. The reason is some pass may legally replace it with another object and then you will have expired pointer. I found out it after implementing first loop version and didn't have enough time to fix it.

I think we can discuss this topic offline - I have a branch that I can share with you where I made the most (I believe) changes to fix that - keep input/output index instead of `vpu::Handle` itself.

If you have arguments it cannot trigger an issue for Faster-RCNN, I think we can leave it as is.

Reply:
We have discussed it offline.

Reply:
Could you describe the result of your offline discussion?

Reply:
This handle to Data is used only inside this pass, so other passes cannot replace it during the execution of this pass.